js/packages/web/src/views/auction/index.tsx

```diff
@@ -289,30 +291,33 @@ const BidLine = (props: {
 }) => {
   const { bid, mint, isCancelled, isLastWinner } = props;
   const { publicKey } = useWallet();
-  const bidder = bid.info.bidderPubkey;
-  const isme = publicKey?.toBase58() === bidder;
+  const bidderPubkeyString = bid.info.bidderPubkey;
+  const isMe = publicKey?.toBase58() === bidderPubkeyString;
 
-  // Get Twitter Handle from address
   const connection = useConnection();
-  const [bidderTwitterHandle, setBidderTwitterHandle] = useState('');
+
+  const [bidderHandle, setBidderHandle] = useState<{
+    handle: string;
+    type: 'pubkey' | 'twitter' | '.sol';
+  }>({
+    handle: shortenAddress(bidderPubkeyString),
+    type: 'pubkey',
+  });
   useEffect(() => {
-    const getTwitterHandle = async (
-      connection: Connection,
-      bidder: StringPublicKey,
-    ): Promise<string | undefined> => {
-      try {
-        const [twitterHandle] = await getHandleAndRegistryKey(
-          connection,
-          toPublicKey(bidder),
-        );
-        setBidderTwitterHandle(twitterHandle);
-      } catch (err) {
-        console.warn(`err`);
-        return undefined;
+    const getBidderHandle = async () => {
+      const twitterHandle = await getTwitterHandle(connection, bidderPubkeyString);
+
+      if (twitterHandle) {
+        setBidderHandle({ handle: twitterHandle, type: 'twitter' });
+      } else {
+        const solDomain = await getSolDomain(connection, bidderPubkeyString);
+        if (solDomain) {
+          setBidderHandle({ handle: solDomain + '.sol', type: '.sol' });
```

Comment:
This is going to slowdown the initial page load. I don't think we can go with this for now.